src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractTypeImportCompletionProvider.cs

```diff
@@ -85,65 +87,83 @@ private async Task AddCompletionItemsAsync(CompletionContext completionContext,
             // which will be used to filter so the provider only returns out-of-scope types.
             var namespacesInScope = GetNamespacesInScope(document, syntaxContext, cancellationToken);
 
+            var tasksToGetCompletionItems = ArrayBuilder<Task<ImmutableArray<CompletionItem>>>.GetInstance();
+
             // Get completion items from current project. 
             var compilation = await project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
-            await typeImportCompletionService.GetTopLevelTypesAsync(project, syntaxContext, HandlePublicAndInternalItem, cancellationToken)
-                .ConfigureAwait(false);
-
-            // Get declarations from directly referenced projects and PEs
+            tasksToGetCompletionItems.Add(typeImportCompletionService.GetTopLevelTypesAsync(project,
+                                                                        syntaxContext,
+                                                                        isInternalsVisible: true,
+                                                                        cancellationToken));
```

Comment:
consider Task.Run here as well, to ensure this kicks off to a threadpool thread.

Reply:
strange indentation.